<launch>
	<arg name="markerSize" default="0.025"/> <!-- in m -->
	<arg name="side" default="left"/>
	<arg name="ref_frame" default="/ECM_base"/> <!-- leave empty and the pose will be published wrt param parent_name -->
	<arg name="global_frame" default="/world"/>
	<arg name="tool_points" default="3"/>
	<arg name="load_reference" default="True"/>

	<include file="$(find easy_handeye)/launch/publish.launch">
		<arg name="namespace_prefix" value="left_camera"/>
		<arg name="eye_on_hand" value="true"/>
	</include>

	<include file="$(find easy_handeye)/launch/publish.launch">
		<arg name="namespace_prefix" value="right_camera"/>
		<arg name="eye_on_hand" value="true"/>
	</include>

	<node pkg="global_frame_registration" name="global_frame_registration" type="global_frame_registration" required="true" output="screen">
        <rosparam command="load" file="$(find global_frame_registration)/config/saras.yaml"/>
		<rosparam command="load" file="$(find global_frame_registration)/config/saras_registrations.yaml"/>
		<param name="global_frame_name" type="string" value="$(arg global_frame)"/>
		<param name="num_points" type="int" value="$(arg tool_points)"/>
		<param name="use_existing" type="bool" value="$(arg load_reference)"/>
	</node>

	<!-- Image must be provided already rectified (seems there's a bug in aruco_ros) -->
	<node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher" output="screen">
		<remap from="/camera_info" to="/endoscope/$(arg side)/camera_info" />
		<remap from="/image" to="/endoscope/$(arg side)/image_rect_color" />
		<param name="image_is_rectified" type="bool" value="True"/>
		<param name="marker_size" value="$(arg markerSize)"/>
		<param name="reference_frame" value="$(arg ref_frame)"/>
		<param name="camera_frame" value="/endoscope/$(arg side)_camera"/>
	</node>
</launch>
